doc/man7/password-handling.pod

```diff
@@ -0,0 +1,89 @@
+=pod
+
+=head1 NAME
+
+password handling - How diverse parts of OpenSSL treat pass phrases
+
+=head1 DESCRIPTION
+
+=head2 The general case
+
+In a modern world with all sorts of character encodings, the treatment of pass
+phrases has become increasingly complex.
+
+The OpenSSL library doesn't treat pass phrases in any special way as a general
+rule, and trusts the application or user to choose a suitable character set
+and stick to that throughout the life of affected objects.
+This means that for an object that was encrypted using a pass phrase encoded in
+ISO-8859-1, that object needs to be decrypted using a pass phrase encoded in
+ISO-8859-1.
+Anything else will have the decryption fail.
+
+=head2 PKCS#12
+
+PKCS#12 is a bit different regarding pass phrase encoding.
+The standard stipulates that the pass phrase shall be encoded as a BMPString,
+i.e. UTF-16.
+
+OpenSSL tries to adapt to this requirements in one of the following manners:
+
+=over 4
+
+=item 1.
+
+Tries to read the pass phrase as UTF-8 and re-encode that to UTF-16, or failing
+that...
+
+=item 2.
+
+Assumes that the pass phrase is encoded in ASCII or ISO-8859-X and
```

Comment:
Not ISO-8859-X. You specifically assume ISO-8859-1. Converting from any *other* ISO-8859-X isn't just a case of interspersing zeroes.

Reply:
However, OpenSSL still treats *anything* that it can't parse as UTF-8 that same way, so for 8-bit charsets other than ISO-8859-1, it till still do the "intersperse zeros" magic (which is incorrect UTF-16, but this is still the current status quo)

Reply:
> -X

It's -X. The only thing that is special about iso8859-1 is that you don't have to open prompt with iso8859-1 input locale, any UTF-8 would actually do. I mean in my last remark I said that you have to stick to original iso8859-X locale to open your PKCS#12 files from 90-s, but reality is that you actually have to only for iso8859>1.

Reply:
Windows uses things like cp1252

Reply:
As [discussed](https://github.com/openssl/openssl/pull/6179#discussion_r186261275) Windows is more or less world of its own when it comes to these things. There is hardly any other choice than to discuss it rather specifically in context of UTF-8 interoperability.